v21.1/deploy-cockroachdb-with-kubernetes-openshift.md

```diff
@@ -0,0 +1,329 @@
+
+## Prerequisites
+
+- A running OpenShift cluster
+
+{{site.data.alerts.callout_info}}
+This article assumes you have already installed the OpenShift Container Platform on a Kubernetes cluster. For details on this, see the [OpenShift documentation](https://docs.openshift.com/container-platform/4.7/installing/index.html).
+{{site.data.alerts.end}}
+
+## Step 1. Create a CockroachDB namespace
+
+1. Create a `cockroachdb` namespace. You will create the CockroachDB cluster in this namespace:
+
+	{{site.data.alerts.callout_info}}
+	`oc` runs `kubectl` commands on OpenShift clusters, using the same syntax.
+	{{site.data.alerts.end}}
+
+	{% include copy-clipboard.html %}
+	~~~ shell
+	oc create namespace cockroachdb
+	~~~
+
+	~~~
+	namespace/cockroachdb created
+	~~~
+
+1.	Set `cockroachdb` as the default namespace:
+
+	{% include copy-clipboard.html %}
+	~~~ shell
+	oc config set-context --current --namespace=cockroachdb
+	~~~
+
+	~~~
+	Context "admin" modified.
+	~~~
+
+	Validate that this was successful:
+
+	{% include copy-clipboard.html %}
+	~~~ shell
+	oc config view --minify | grep namespace:
+	~~~
+
+	~~~
+	namespace: cockroachdb
+	~~~
+
+	This lets you issue `oc` commands without having to specify the namespace each time.
+
+## Step 2. Install the Operator
+
+1. Navigate to your OpenShift web console and click on **OperatorHub**.
+
+1. Enter "cockroach" in the search filter. There are two tiles called **CockroachDB Operator**. Find the tile _without_ the `Marketplace` label (which requires a subscription).
+
+	<img src="{{ 'images/v21.1/cockroachdb-operator-openshift.png' | relative_url }}" alt="OpenShift OperatorHub" style="border:1px solid #eee;max-width:100%" />
+
+	Click the **CockroachDB Operator** tile and then **Install**.
+
+1. On the **Install Operator** page, select `cockroachdb` in the **Installed Namespace** dropdown and click **Install**.
+
+1. Validate that the Operator is running:
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    $ oc get pods
+    ~~~
+
+	~~~
+	NAME                                  READY   STATUS    RESTARTS   AGE
+	cockroach-operator-65c4f6df45-h5r5n   1/1     Running   0          51s
+	~~~
+
+### Step 3. Start CockroachDB
+
+1. When the Operator is ready, click **View Operator** to navigate to the **Installed Operators** page.
+
+1. In the **CockroachDB Operator** tile, click **Create instance**.
+
+	<img src="{{ 'images/v21.1/cockroachdb-operator-instance-openshift.png' | relative_url }}" alt="OpenShift OperatorHub" style="border:1px solid #eee;max-width:100%" />
+
+1. Make sure **CockroachDB Version** is set to an Operator-supported version. For a list of compatible image names, see the [Operator deployment definition](https://github.com/cockroachdb/cockroach-operator/blob/master/manifests/operator.yaml) on GitHub.
```

Comment:
This is awkward, and realistically we want to direct people to use the cockroachDBVersion api and not the image.name anyway on Red Hat.  We should instead maybe say something like `Please make sure that cockroachDBVersion is set to a valid version of CockroachDB, like v21.1.0 or v20.2.8.  This will use the official CockroachDB images that are certified by Red Hat.  Alternatively, customers can also set spec.image.name to pull a CockroachDB image from a custom location.

Reply:
fwiw the cockroachDBVersion api was built after I wrote the original readme so I should have clarified this with you earlier - sorry about that!